pkg/reconciler/autoscaling/reconciler.go

```diff
@@ -138,18 +138,30 @@ func (c *Base) ReconcileMetric(ctx context.Context, pa *pav1alpha1.PodAutoscaler
 }
 
 // UpdateStatus updates the status of the given PodAutoscaler.
-func (c *Base) UpdateStatus(desired *pav1alpha1.PodAutoscaler) (*pav1alpha1.PodAutoscaler, error) {
-	pa, err := c.PALister.PodAutoscalers(desired.Namespace).Get(desired.Name)
-	if err != nil {
-		return nil, err
-	}
-	// If there's nothing to update, just return.
-	if reflect.DeepEqual(pa.Status, desired.Status) {
-		return pa, nil
-	}
-	// Don't modify the informers copy
-	existing := pa.DeepCopy()
-	existing.Status = desired.Status
+func (c *Base) UpdateStatus(desired *pav1alpha1.PodAutoscaler) error {
+	return reconciler.RetryUpdateConflicts(func(i int) error {
+		var (
+			existing *pav1alpha1.PodAutoscaler
+			err      error
+		)
+		// The first iteration tries to use the informer's state.
+		if i == 0 {
+			existing, err = c.PALister.PodAutoscalers(desired.Namespace).Get(desired.Name)
+			existing = existing.DeepCopy()
+		} else {
+			existing, err = c.ServingClientSet.AutoscalingV1alpha1().PodAutoscalers(desired.Namespace).Get(desired.Name, metav1.GetOptions{})
+		}
+		if err != nil {
+			return err
+		}
+
+		// If there's nothing to update, just return.
+		if reflect.DeepEqual(existing.Status, desired.Status) {
```

Comment:
apiequality.Semantic?

Reply:
I realize this wasn't in the prior code - feel free to leave this as a separate PR or I can do it

Reply:
Yeah, let's leave it separate. Was wondering the same while doing it though.

Reply:
`kmp.Diff` ? :)

Reply:
As mentioned, won't fix in this PR. I don't want to inadvertedly and subtly break things in this PR which is already kinda invasive. Let's make that a separate change please.

Reply:
+1 to doing separately.